src/main/java/io/candydoc/ddd/SubdomainType.java

```diff
@@ -0,0 +1,7 @@
+package io.candydoc.ddd;
+
+public enum SubdomainType {
+  CoreDomain,
```

Comment:
Name of the subdomain type is inconsistent with other type.

Why is this one no called `CoreSubdomain` ?
Why do we need to suffix this although the type is about subdomains ?

It's like having an `Orientation` type and with `NorthOrientation`. `Orientation` doesn't provide any valuable information in `NorthOrientation` as it is already contextualised within the type.